src/lib/statuses.ts

```diff
@@ -29,53 +54,59 @@ export const getStatuses = (
     isHintMode = loaded.isHintMode
   }
 
+  function updateCharObjectKey(letter: string, value: CharStatus) {
+    // Sets a new status of a key, only if the new status is more important than a previous status
+    // i.e. 'present' does not override 'correct', if the character appears twice
+    const currentStatusPriority = charStatusRanking[charObj[toHiragana(letter)]]
+    const newStatusPriority = charStatusRanking[value]
+    if (!currentStatusPriority || newStatusPriority > currentStatusPriority) {
+      charObj[toHiragana(letter)] = value
+    }
+  }
+
   guesses.forEach((word) => {
     unicodeSplit(word).forEach((letter, i) => {
-
       if (isHintMode) {
         vowelStatusKana.forEach((kana) => {
           if (kana.includes(letter) && kana.includes(splitSolution[i])) {
-            //make status close
-            return (charObj[toHiragana(letter)] = 'vowel')
+            updateCharObjectKey(letter, 'vowel')
           }
         })
 
         consonantStatusKana.forEach((kana) => {
           if (kana.includes(letter) && kana.includes(splitSolution[i])) {
-            //make status close
-            return (charObj[toHiragana(letter)] = 'consonant')
+            updateCharObjectKey(letter, 'consonant')
           }
         })
 
         closeStatusKana.forEach((kana) => {
           if (kana.includes(letter) && kana.includes(splitSolution[i])) {
-            //make status close
-            return (charObj[toHiragana(letter)] = 'close')
+            updateCharObjectKey(letter, 'close')
           }
         })
       }
 
-      if (!splitSolution.includes(letter) && !['vowel', 'consonant', 'present', 'close', 'correct'].includes(charObj[toHiragana(letter)])) {
-        // make status absent
-        return (charObj[toHiragana(letter)] = 'absent')
+      if (!splitSolution.includes(letter)) {
+        updateCharObjectKey(letter, 'absent')
       }
 
-      if (letter === splitSolution[i]) {
-        //make status correct
-        return (charObj[toHiragana(letter)] = 'correct')
+      if (splitSolution.includes(letter)) {
+        updateCharObjectKey(letter, 'present')
       }
 
-      if (splitSolution.includes(letter) && !['close', 'correct'].includes(charObj[toHiragana(letter)])) {
-        //make status present
-        return (charObj[toHiragana(letter)] = 'present')
+      if (letter === splitSolution[i]) {
+        updateCharObjectKey(letter, 'correct')
```

Comment:
None of these if statements return early now. It could be refactored so that we check in decreasing priority order (e.g. start with correct, close, present, etc.). In this case it would be safe to return early as a later statement could never result in an override. However I've just left the order as is and removed the returns.